docs/experiments-duot5-tpu.md

```diff
@@ -0,0 +1,242 @@
+- `qrels.dev.small.tsv`: 7,437 pairs of query relevant passage ids from the MS MARCO dev set.
+- `collection.tar.gz`: All passages (8,841,823) in the MS MARCO passage corpus. In this tsv file, the first column is the passage id, and the second is the passage text.
+
+A more detailed description of the data is available [here](https://github.com/castorini/duobert#data-and-trained-models).
+
+Let's start.
+```
+cd ${DATA_DIR}
+wget https://www.dropbox.com/s/hq6xjhswiz60siu/queries.dev.small.tsv
+wget https://www.dropbox.com/s/5t6e2225rt6ikym/qrels.dev.small.tsv
+wget https://www.dropbox.com/s/m1n2wf80l1lb9j1/collection.tar.gz
+tar -xvf collection.tar.gz
+rm collection.tar.gz
+cd ../../
+```
+
+As a sanity check, we can evaluate the second-stage retrieved documents using the official MS MARCO evaluation script.
+We  choose one of the monoT5-base run file to rerank with duoT5-base and the monoT5-3B run file to rerank with duoT5-3B.
+```
+export MODEL_NAME=<base or 3B>
+python tools/eval/msmarco_eval.py ${DATA_DIR}/qrels.dev.small.tsv ${DATA_DIR}/run.monot5_${MODEL_NAME}.dev.tsv
+```
+
+In the case of monoT5-3B, the output should be:
+
+```
+#####################
+MRR @10: 0.3983799517896949
+QueriesRanked: 6980
+#####################
+```
+
+In the case of monoT5-base, the output should be:
+
+```
+#####################
+MRR @10: 0.38160657433938283
+QueriesRanked: 6980
+#####################
+```
+
+Then, we prepare the query-doc0-doc1 pairs in the duoT5 input format.
+```
+python pygaggle/data/create_msmarco_duot5_input.py --queries ${DATA_DIR}/queries.dev.small.tsv \
+                                      --run ${DATA_DIR}/run.monot5_${MODEL_NAME}.dev.tsv \
+                                      --corpus ${DATA_DIR}/collection.tsv \
+                                      --t5_input ${DATA_DIR}/query_docs_triples.dev.small.txt \
+                                      --t5_input_ids ${DATA_DIR}/query_docs_triple_ids.dev.small.tsv \
+                                      --top_k 50
+```
+We will get two output files here:
+- `query_docs_triples.dev.small.txt`: The query-doc0-doc1 triples for duoT5 input.
+- `query_docs_triple_ids.dev.small.tsv`: The `query_id`s,`doc_id_0`s, and `doc_id_1`s that map to the query-doc0-doc1 triples. We will use this to map query-doc0-doc1 triples to their corresponding duoT5 output scores.
+
+The files are made available in our [bucket](https://console.cloud.google.com/storage/browser/castorini/duot5/data).
+
+Note that there might be a memory issue if the duoT5 input file is too large for the memory in the instance. We thus split the input file into multiple files.
+
+```
+split --suffix-length 3 --numeric-suffixes --lines 500000 ${DATA_DIR}/query_docs_triples.dev.small.txt ${DATA_DIR}/query_docs_triples.dev.small.txt
+```
+
+For `query_docs_triples.dev.small.txt`, we will get 9 files after split. i.e. (`query_docs_triples.dev.small.txt000` to `query_docs_triples.dev.small.txt034`).
```

Comment:
We say 9 files here but below we iterate in 35 files ({000..034}. Could please also confirm if `--lines 500000` is correct?

Reply:
Oops should say 35 files, will fix it!